lib/manager/npm/post-update/index.ts

```diff
@@ -438,6 +444,11 @@ export async function getAdditionalFiles(
       uri = validateUrl(uri) ? uri.replace(/^https?:/, '') : `//${uri}/`;
       if (hostRule.token) {
         const key = hostRule.authType === 'Basic' ? '_auth' : '_authToken';
+
+        // Yarn 2
+        env.YARN_NPM_REGISTRY_SERVER = uri;
```

Comment:
does this do what I hope it does? if it sets the environment variables on the running machine, this should all work

Reply:
nope, this will set the last found hostrule with a token. This will only work if there is only one hostrule.